tools/tcp_sources/tcp_netstat.c

```diff
@@ -0,0 +1,194 @@
+/*
+ * FreeRTOS+TCP V2.3.0
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * http://aws.amazon.com/freertos
+ * http://www.FreeRTOS.org
+ */
+
+/* Standard includes. */
+#include <stdint.h>
+#include <stdio.h>
+
+/* FreeRTOS includes. */
+#include "FreeRTOS.h"
+#include "task.h"
+#include "queue.h"
+#include "semphr.h"
+
+/* FreeRTOS+TCP includes. */
+#include "FreeRTOS_IP.h"
+#include "FreeRTOS_Sockets.h"
+#include "FreeRTOS_IP_Private.h"
+
+#include "tcp_netstat.h"
+
+extern List_t xBoundUDPSocketsList;
+
+#if ( ipconfigUSE_TCP == 1 )
+    extern List_t xBoundTCPSocketsList;
+#endif /* ipconfigUSE_TCP == 1 */
+
+IOCounters_t xInputCounters, xOutputCOunters;
+
+BaseType_t vGetMetrics( MetricsType_t * pxMetrics )
+{
+    BaseType_t xResult = 0;
+
+    /* Show a simple listing of all created sockets and their connections */
+    const ListItem_t * pxIterator;
+    uint16_t usLastPort;
+
+    memset( pxMetrics, 0, sizeof *pxMetrics );
+
+    pxMetrics->xInput = xInputCounters;
+    pxMetrics->XOutput = xOutputCOunters;
```

Comment:
```suggestion
    pxMetrics->XOutput = xOutputCounters;
```

Reply:
Not only a hanging shift, also copy/pasted the typo :-)